### PR TITLE
#384 Implement upload images functionality

### PR DESCRIPTION
**Pull request by Lvyshnevska**
Functionality of uploading banner/logo images has been implemented. 
Toast messages about successfully saved data/images have been added to all components on Edit Profile page.

### CODE REVIEW

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
             <div className={css['fields__label']}>
                 {props.requredField && <label className={css['fields__label--required']}>*</label>}
                 <label
-                    className={`${css['fields__label--text']} ${!props.requredField && css['fields__field--notrequired']}`}
-                >
+                    className={classNames(css['fields__label--text'], {[css['fields__field--notrequired']] : !props.requiredField})}>
                     {props.label}
                 </label>
+                {props.name === 'banner_image' && props.value &&
+                    (<span className={css['upload-file__delete--wrapper']} onClick={() => (props.onDeleteImage(props.name))}>
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js:17]: Could we use more semantic element for like `button` or at least add role="button"

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js [hunk 2]
```diff
@@ -1,39 +1,64 @@
...
+                        <img className={css['upload-file__delete--trashBin']} src={`${process.env.REACT_APP_PUBLIC_URL}/profilepage/Vectordelete.png`} alt="" />
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js:18]: We can add alt here as `delete icon` or else

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js [hunk 3]
```diff
@@ -1,39 +1,64 @@
...
+                                Зображення для карток
+                            </span>
+                            <div className={css['upload-file__banner-image--card']} style={backgroundImage} />
+                        </div>
+                    </>
+                )}
+                {props.name === 'logo_image' && props.value && (
+                    <div className={css['upload-file__wrapper--logo']}>
+                        <div className={css['upload-file__logo']} style={backgroundImage}/>
+                        <span className={css['upload-file__delete--wrapper']} onClick={() => (props.onDeleteImage(props.name))}>
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js:56]: the same as above

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.module.css [hunk 4]
```diff
@@ -117,21 +120,88 @@
...
-    text-align: left;
+    letter-spacing: -0.14px;
 }
 
-.upload-file__filename--trashBin {
-    padding-top: 1px;
+.upload-file__banner-image--page {
+    width: 420px;
+    height: 55.556px;
+    border: 1px solid #C3C3C3;
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.module.css:159]: `global.css` has not merged so far but can use a variable with a default value

File: FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js [hunk 5]
```diff
@@ -165,6 +176,80 @@ const GeneralInfo = (props) => {
...
+                        setLogoImage(mainProfile.logo_image);
+                    }
+                    data[imageKey] === null
+                        ? toast.success(imageKey === 'banner_image' ? 'Банер видалено з профілю' : 'Логотип видалено з профілю')
+                        : toast.success(imageKey === 'banner_image' ? 'Банер успішно додано у профіль' : 'Логотип успішно додано у профіль');
+                } else if (response.status === 400) {
+                    toast.error('Не вдалося завантажити банер/лого, сталася помилка');
+                }
+            } catch (error) {
+                console.error('Error uploading image:', error);
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js:209]: I think we should show toast message here as well

File: FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js [hunk 6]
```diff
@@ -165,6 +176,80 @@ const GeneralInfo = (props) => {
...
+                } else {
+                    setLogoImageError(null);
+                    setLogoImage(file);
+                    setProfile((prevState) => {
+                        const newState = { ...prevState, [e.target.name]: imageUrl };
+                        return newState;
+                    });
+                }
+            }
+        }
```
Comment (popovycholeg) [inline FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js:242]: So many if/else clauses, could we try to break them into a few functions?

File: FrontEnd/src/components/basicPage/BasicPage.jsx [hunk 7]
```diff
@@ -26,7 +27,7 @@ import TermsAndConditions from '../terms-and-conditions-app/terms_conditions/Ter
 import { useAuth } from '../../hooks';
 import { useUser } from '../../hooks';
 import { Search } from '../SearchPage/Search';
-import classes from './BasicPage.module.css';
+import './customToastStyles.css';
```
Comment (popovycholeg) [inline FrontEnd/src/components/basicPage/BasicPage.jsx:30]: use module css

**Review by popovycholeg — APPROVED** [review]
With some non-critical comments

File: FrontEnd/src/components/basicPage/BasicPage.jsx (hunk 7 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/basicPage/BasicPage.jsx:30]: Overridden toast styles are not applying if use module css.

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.module.css (hunk 4 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.module.css:159]: I've added this color according to design on Figma . And there is information - this color is not tied to a style or variable in design. So I can't replace it with a variable.

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js (hunk 1 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js]: Done

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js (hunk 2 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js]: Done

File: FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js (hunk 3 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/FormFields/ImageField.js]: Done

File: FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js (hunk 5 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js:209]: Done

File: FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js (hunk 6 above)
Comment (Lvyshnevska) [reply FrontEnd/src/components/ProfilePage/FormComponents/GeneralInfo.js]: Done

**Review by Shushunya — APPROVED** [review]

**Pull request merged**